Snek.cs

```diff
@@ -1,131 +1,132 @@
     // starting position and direction of the snake
-    public int snakeRow = 10;
-    public int snakeCol = 10;
+    public Cord[] snakeCords;
     public Direction direction = Direction.Right;
 
     // direction in which it can move
     int[,] D = { { 0, 1 }, { 1, 0 }, { 0, -1 }, { -1, 0 } };
 
-    // storing the last position in the arrays based on top score
-    public int[] prevLocRow;
-    public int[] prevLocCol;
-
     public Snek(int topScore)
     {
-        prevLocRow = new int[topScore];
-        prevLocCol = new int[topScore];
+        snakeCords = new Cord[topScore + 1];
+        snakeCords[0] = new() { X = 10, Y = 10 };
+    }
+
+    // snek properties for row and col 
+    public Cord Head
+    {
+        get { return snakeCords[0]; }
+        set { snakeCords[0] = value; }
     }
 
     // methods now
-    public void Move(int gridRows, int gridCols)
+    public void Move(Grid grid)
     {
         //keeps track of all of the snake
-        int a = prevLocRow.Length - 2;
+        int a = snakeCords.Length - 2;
         while (a >= 0)
 
         {
-            prevLocRow[a + 1] = prevLocRow[a];
-            prevLocCol[a + 1] = prevLocCol[a];
+            snakeCords[a + 1] = snakeCords[a];
             a = a - 1;
         }
 
-        prevLocRow[0] = snakeRow;
-        prevLocCol[0] = snakeCol;
-        snakeRow += D[(int)direction, 0];
-        snakeCol += D[(int)direction, 1];
+        //endless faff of using structs fml 
+        Cord newHead = Head;
+
+        newHead.X += D[(int)direction, 0];
+        newHead.Y += D[(int)direction, 1];
 
         // wrapping around situ
-        if (snakeRow >= gridRows){snakeRow = 0;}
-        if (snakeRow < 0){snakeRow = gridRows - 1;}
-        if (snakeCol >= gridCols){snakeCol = 0;}
-        if (snakeCol < 0){snakeCol = gridCols - 1;}
+        if (newHead.X >= grid.cols) { newHead.X = 0; }
+        if (newHead.X < 0) { newHead.X = grid.cols - 1; }
+        if (newHead.Y >= grid.rows) { newHead.Y = 0; }
+        if (newHead.Y < 0) { newHead.Y = grid.rows - 1; }
 
+        Head = newHead;
     }
 
     public void SetDirection(Direction direction)
     {
         this.direction = direction;
     }
 
-    public bool Overlaps(int row, int col, int score, bool skipHead = false)
+    public bool Overlaps(Cord newSnakeCords, int score, bool skipHead = false)
     {
         // 1. Check the head 
-        if (!skipHead && snakeRow == row && snakeCol == col)
+        if (!skipHead && Head.Equals(newSnakeCords))
         {
             return true;
         }
         else    // 2. Check the rest of the body through loop with PrevLoc
         {
-            for (int i = 0; i < score; i++)
+            for (int i = 1; i < score; i++)
             {
-                if (prevLocRow[i] == row && prevLocCol[i] == col)
+                if (snakeCords[i].Equals(newSnakeCords))
                 {
                     return true;
                 }
             }
             return false;
         }
```

Comment:
OK here's a bit of an opportunity to clean up. Right now, you have a separate condition for the head.

And at the bottom you loop through the list of things and do the same thing. 

So skipHead really only dictates if you loop from 0 or from 1. If skiphead is true you loop through starting from 1 if false, you start from 0. 

Clean this up by trying to make it work with just one loop. Make sure to commit all your work before you try this so if yo ucan't do you can revert.